internal/metrics/metrics.go

```diff
@@ -127,3 +130,58 @@ func (s *ScheduleTimer) Mark(ctx context.Context, role, outcome string) {
 	metrics.FromContext(ctx).Histogram(
 		schedulingRetryTime, sparkRoleTag, outcomeTag, s.instanceGroupTag, s.retryTag).Update(now.Sub(s.lastSeenTime).Nanoseconds())
 }
+
+func ReportCrossZoneMetric(ctx context.Context, driverNodeName string, executorNodeNames []string, nodes []*v1.Node) {
+	executorNodesSet := make(map[string]interface{})
+	for _, n := range executorNodeNames {
+		executorNodesSet[n] = nil
+	}
+
+	zonesCounter := make(map[string]int64)
+	for _, n := range nodes {
+		if _, ok := executorNodesSet[n.Name]; ok {
+			executorZone, err := getNodeZone(n)
+			if err != nil {
+				return
+			}
+			increment(zonesCounter, executorZone)
+		} else if n.Name == driverNodeName {
+			driverZone, err := getNodeZone(n)
+			if err != nil {
+				return
+			}
+			increment(zonesCounter, driverZone)
+		}
+	}
+
+	metrics.FromContext(ctx).Histogram(crossAzTraffic).Update(crossZoneTraffic(zonesCounter))
+}
+
+func increment(counter map[string]int64, zoneName string) {
+	if _, ok := counter[zoneName]; !ok {
+		counter[zoneName] = 0
+	}
+	counter[zoneName] += 1
+}
+
+func getNodeZone(node *v1.Node) (string, error) {
```

Comment:
because you don't handle the returned error, I think you can remove this function,
and above just do
```
driverZone, ok := node.Labels[nodeZoneLabel]
if !ok {
  // log warn here
}
```

Reply:
yup, added a warn log